packages/story-editor/src/components/canvas/videoCaptions/videoCaptionsLayer.js

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import { useEffect, useState } from '@web-stories-wp/react';
+import styled from 'styled-components';
+import { useFeature } from 'flagged';
+
+/**
+ * Internal dependencies
+ */
+import { useStory, useCanvas } from '../../../app';
+import { Layer, PageArea } from '../layout';
+import TrackRenderer from './trackRenderer';
+
+const CaptionsLayer = styled(Layer)``;
+
+const CaptionsPageArea = styled(PageArea)`
+  position: absolute;
+  pointer-events: none;
+`;
+
+const CaptionsCanvas = styled.div`
+  background: transparent;
+  border-radius: 5px;
+  position: absolute;
+  z-index: 2;
+  transform: translateZ(0);
+  top: 0;
+  left: 0;
+  bottom: 0;
+  right: 0;
+  cursor: default;
+`;
+
+function VideoCaptionsLayer() {
+  const isFeatureEnabled = useFeature('customVideoCaptionsInEditor');
+
+  const { isEditing } = useCanvas(({ state: { isEditing } }) => ({
+    isEditing,
+  }));
+
+  const videoElement = useStory(({ state }) => {
+    const { currentPage, selectedElementIds } = state;
+
+    if (selectedElementIds.length !== 1) {
+      return null;
+    }
+
+    return (currentPage?.elements || [])
```

Comment:
If we're only checking for the single selected element anyway, is it necessary to filter all the elements on the current page? Maybe just use `selectedElements[0]` to check if it's a video and has `tracks.length`? WDYT?

Reply:
Totally 🤦 

I used `elements` first before realizing that one can only play the selected element anyway.